### PR TITLE
[DOCS] Update assembly.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/assembly.rst [hunk 1]
```diff
@@ -47,13 +47,20 @@ these curly braces, you can use the following (see the later sections for more d
...
- - assignments in functional style, e.g. ``x := add(y, 3)``
+ - identifiers (assembly-local variables and externals if used as inline assembly), e.g. ``add(3, x)``, ``sstore(x_slot, 2)``
+ - assignments, e.g. ``x := add(y, 3)``
  - blocks where local variables are scoped inside, e.g. ``{ let x := 3 { let y := add(x, 1) } }``
 
+The following features are only available for standalone assembly.
+Standalone assembly is supported for backwards compatibility but is not documented
+here anymore.
+
+ - direct stack contral via dup1, swap1, ...
```
Comment (axic) [inline docs/assembly.rst]: contral -> control

But I'd format this paragraph a bit more nicer.

Comment (chriseth) [reply docs/assembly.rst]: Any suggestions? Do you want to take this over?

File: docs/assembly.rst [hunk 2]
```diff
@@ -334,38 +341,39 @@ Literals
 You can use integer constants by typing them in decimal or hexadecimal notation and an
 appropriate ``PUSHi`` instruction will automatically be generated. The following creates code
 to add 2 and 3 resulting in 5 and then computes the bitwise and with the string "abc".
```
Comment (erak) [inline docs/assembly.rst:349]: Maybe combine both sentences? "...resulting in 5, then computes the bitwise ``and`` with the string "abc" and assignes the final value to a local variable called ``x``"

File: docs/assembly.rst (hunk 1 above)
Comment (erak) [reply docs/assembly.rst]: Perhaps it's enough to pre-format the instructions using back-ticks. Should we move the note about deprecation of instructions-style assignments up here?

**Comment by erak** [comment]
Updated.

**Comment by chriseth** [comment]
Rebased to fix the tests.

**Pull request merged**